FlowCrypt/Controllers/MessageList Extension/MsgListViewController.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  MsgListViewController.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 23/12/2019.
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import UIKit
+
+protocol MsgListViewController {
+    func open(with message: InboxRenderable, path: String)
+
+    func getUpdatedIndex(for message: InboxRenderable) -> Int?
+    func updateMessage(isRead: Bool, at index: Int)
+    func removeMessage(at index: Int)
+}
+
+extension MsgListViewController where Self: UIViewController {
+    func open(with message: InboxRenderable, path: String) {
+        switch message.wrappedType {
+        case .message(let message):
+            openMsgElseShowToast(with: message, path: path)
+        case .thread(let thread):
+            openThread(with: thread)
+        }
+    }
+
+    // TODO: uncomment in "sent message from draft" feature
+    private func openDraft(with message: Message) {
+        guard let email = DataService.shared.email else { return }
+
+        Task {
+            let controller = await ComposeViewController(email: email)
+            await controller.updateWithMessage(message: message)
+            await navigationController?.pushViewController(controller, animated: true)
+        }
+    }
+
+    private func openMsgElseShowToast(with message: Message, path: String) {
+        if message.size ?? 0 > GeneralConstants.Global.messageSizeLimit {
+            showToast("Messages larger than 5MB are not supported yet")
+        } else {
+            let messageInput = MessageViewController.Input(
+                objMessage: message,
+                bodyMessage: nil,
+                path: path
+            )
+            let msgVc = MessageViewController(input: messageInput) { [weak self] (action, message) in
+                self?.handleMessageOperation(with: message, action: action)
+            }
+            navigationController?.pushViewController(msgVc, animated: true)
+        }
+    }
+
+    private func openThread(with thread: MessageThread) {
+        guard !thread.messages.contains(where: { ($0.size ?? 0) > GeneralConstants.Global.messageSizeLimit }) else {
```

Comment:
@tomholub 
Also we use `GeneralConstants.Global.messageSizeLimit` here

Reply:
See https://github.com/FlowCrypt/flowcrypt-ios/commit/03e6f3551bd15dc990e97c0c55ea5aa9d670d794